File: 2_pointers/isPalindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string res;
        for(char a:s){
            if ((a >= 'a' && a <= 'z') || 
                (a >= 'A' && a <= 'Z') || 
                (a >= '0' && a <= '9')) {
                res += tolower(a);
            }
        }
        int r = 0;
        int l = res.size()-1;
        while(r<l){
            if(res[r] != res[l]) return false;
            r++;
            l--;
        }

        return true;
    }
};

File: 2_pointers/isPalindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        res = ""
        for a in s:
            if a.isalnum():
                res += a.lower()

        r = 0
        l = len(res)-1
        while  r<l :
            if res[r] != res[l]:
                return False
            r+=1
            l-=1

        return True

        